saleor/graphql/invoice/mutations.py

```diff
@@ -61,15 +61,28 @@ def perform_mutation(cls, _root, info, **data):
         )
         cls.clean_order(order)
 
-        invoice = models.Invoice.objects.create(order=order, number=data.get("number"),)
-
-        info.context.plugins.invoice_request(
-            order=order, invoice=invoice, number=data.get("number")
+        shallow_invoice = models.Invoice.objects.create(
+            order=order, number=data.get("number"),
+        )
+        invoice = info.context.plugins.invoice_request(
+            order=order, invoice=shallow_invoice, number=data.get("number")
         )
+
+        if invoice.status == JobStatus.SUCCESS:
+            order.events.create(
+                type=OrderEvents.INVOICE_GENERATED,
+                order=order,
+                user=info.context.user,
+                parameters={"invoice_number": invoice.number},
+            )
+        else:
+            order.events.create(
```

Comment:
In almost all places we use a helper to create an event for example.
`events.invoice_requested_event`

Reply:
That would be better for consistency.